src/WooCommerceCheckout.php

```diff
@@ -28,6 +28,36 @@ public static function init() {
       add_filter('woocommerce_checkout_fields', __CLASS__ . '::addConfirmationEmailCheckoutField');
       add_action('woocommerce_checkout_process', __CLASS__ . '::checkConfirmationEmailField');
     }
+
+    add_filter( 'woocommerce_form_field', __CLASS__ . '::woocommerceFormField', 10, 4 );
+  }
+
+  /**
+   * Adds inline error messages to checkout fields when needed.
+   *
+   * @implements woocommerce_form_field
+   *
+   * @return string
+   */
+  public static function woocommerceFormField($field, $key, $args, $value) {
+    if (strpos($field, '</label>') === FALSE) {
+      return $field;
+    }
+
+    $error = [];
+    if ($args['required'] ?? FALSE) {
+      $errorMessage = sprintf(__('%s is a required field.', 'woocommerce'), $args['label']);
```

Comment:
Local variables should be in lower_snake_case while class properties are in lowerCamelCase.
```suggestion
      $error_message = sprintf(__('%s is a required field.', 'woocommerce'), $args['label']);
```

Reply:
this variable differentiation is also a drupal convention? or why is it like this?

Reply:
Yes https://www.drupal.org/docs/develop/standards/coding-standards#s-functions-and-variables we agreed on using the different cases through the files. But I agree we should rather stick to a single convention instead of mixing it depending on the type. E.g. use lowerCamelCase. I think this needs to be discussed mainly with @sun.

Reply:
Alright, worth to mention thanks